src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -61,33 +66,78 @@ public String getSendAs() {
+        return notifyAborted;
+    }
+
+    public boolean getNotifyFailure() {
+        return notifyFailure;
+    }
+
+    public boolean getNotifyNotBuilt() {
+        return notifyNotBuilt;
+    }
+
+    public boolean getNotifyUnstable() {
+        return notifyUnstable;
+    }
+
+    public boolean getNotifyBackToNormal() {
+        return notifyBackToNormal;
+    }
+
+    public boolean includeTestSummary() {
+        return includeTestSummary;
+    }
+
+    public boolean getNotifyRepeatedFailure() {
+        return notifyRepeatedFailure;
+    }
+
     @DataBoundConstructor
-    public SlackNotifier(final String teamDomain, final String authToken, final String room, String buildServerUrl, final String sendAs) {
+    public SlackNotifier(final String teamDomain, final String authToken, final String room, final String buildServerUrl,
+                         final String sendAs, final boolean startNotification, final boolean notifyAborted, final boolean notifyFailure,
+                         final boolean notifyNotBuilt, final boolean notifySuccess, final boolean notifyUnstable, final boolean notifyBackToNormal,
+                         final boolean notifyRepeatedFailure, final boolean includeTestSummary, final boolean showCommitList) {
         super();
         this.teamDomain = teamDomain;
         this.authToken = authToken;
         this.buildServerUrl = buildServerUrl;
         this.room = room;
         this.sendAs = sendAs;
+        this.startNotification = startNotification;
+        this.notifyAborted = notifyAborted;
+        this.notifyFailure = notifyFailure;
+        this.notifyNotBuilt = notifyNotBuilt;
+        this.notifySuccess = notifySuccess;
+        this.notifyUnstable = notifyUnstable;
+        this.notifyBackToNormal = notifyBackToNormal;
+        this.notifyRepeatedFailure = notifyRepeatedFailure;
+        this.includeTestSummary = includeTestSummary;
+        this.showCommitList = showCommitList;
     }
 
     public BuildStepMonitor getRequiredMonitorService() {
         return BuildStepMonitor.BUILD;
     }
 
-    public SlackService newSlackService(String teamDomain, String token, String projectRoom) {
-        // Settings are passed here from the job, if they are null, use global settings
-        if (teamDomain == null) {
-            teamDomain = getTeamDomain();
-        }
-        if (token == null) {
-            token = getAuthToken();
-        }
-        if (projectRoom == null) {
-            projectRoom = getRoom();
-        }
-
-        return new StandardSlackService(teamDomain, token, projectRoom);
+    public SlackService newSlackService() {
+        String teamDomain = StringUtils.isEmpty(this.teamDomain) ? getDescriptor().getTeamDomain() : this.teamDomain;
+        String authToken = StringUtils.isEmpty(this.authToken) ? getDescriptor().getToken() : this.authToken;
+        String room = StringUtils.isEmpty(this.room) ? getDescriptor().getRoom() : this.room;
```

Comment:
I agree with your opinion that should keep the code style consistent throughout the project.
I'll fix.